Kubernetes/legos/k8s_get_expiring_certificates/k8s_get_expiring_certificates.py

```diff
@@ -0,0 +1,73 @@
+##
+# Copyright (c) 2023 unSkript, Inc
+# All rights reserved.
+##
+from pydantic import BaseModel, Field
+from typing import Optional, Tuple
+import base64
+from kubernetes import client
+from kubernetes.client.rest import ApiException
+from datetime import datetime, timedelta
+from unskript.legos.kubernetes.k8s_kubectl_command.k8s_kubectl_command import k8s_kubectl_command
+
+
+class InputSchema(BaseModel):
+    namespace: Optional[str] = Field(
+        default='',
+        title='Namespace',
+        description='K8s Namespace. Default- all namespaces')
+    expiring_threshold: Optional[int] = Field(
+        default=90,
+        title='Expiration Threshold (in days)',
+        description='Expiration Threshold of certificates (in days). Default- 90 days')
+
+
+def k8s_get_expiring_certificates_printer(output):
+    if output is None:
+        return
+    print(output)
+
+def k8s_get_expiring_certificates(handle, namespace:str='', expiring_threshold:int=90) -> Tuple:
+    """
+    Get the expiring certificates for a K8s cluster.
+
+    Args:
+        handle: Object of type unSkript K8S Connector
+        namespace (str): The Kubernetes namespace where the certificates are stored.
+        expiration_threshold (int): The threshold (in days) for considering a certificate as expiring soon.
+
+    Returns:
+        tuple: Status, a list of expiring certificate names.
+    """
+    result = []
+    all_namespaces = [namespace]
+    secrets = []
+    cmd = "kubectl get ns  --no-headers -o custom-columns=':metadata.name'"
+    if namespace is None or len(namespace)==0:
+        kubernetes_namespaces = k8s_kubectl_command(handle=handle,kubectl_command=cmd )
+        replaced_str = kubernetes_namespaces.replace("\n"," ")
+        stripped_str = replaced_str.strip()
+        all_namespaces = stripped_str.split(" ")
+    coreApiClient = client.CoreV1Api(api_client=handle)
+    try:
+        coreApiClient.read_namespace_status(namespace, pretty=True)
+    except ApiException as e:
+        raise e
+    for n in all_namespaces:
+        secrets.append(coreApiClient.list_namespaced_secret(n).items)
```

Comment:
@shloka-bhalgat-unskript do we handle pagination for k8s apis, if not, can we track it with an issue and fix all apis. [This](https://cloudnativenow.com/topics/cloudnativeplatforms/introduction-to-using-the-kubernetes-rest-api/) article talks about how to do pagination

Reply:
Sure will open a separate ticket to fix other legos using the client without pagination. I have added it for this lego.